R/xgboost-deployment.R

```diff
@@ -25,7 +24,26 @@
 #' F leads to removal of rows containing NULLs.
 #' @param debug Provides the user extended output to the console, in order
 #' to monitor the calculations throughout. Use T or F.
-#' @return Returns a dataframe containing the grain column, the top 3 probabilities for each row, and the classes associated with those probabilities.
+#' @param cores Number of cores you'd like to use.  Defaults to 2.
+#' @param modelName Optional string. Can specify the model name. If used, you must load the same one in the deploy step.
+#' @section Methods: 
+#' The above describes params for initializing a new XGBoostDeployment class with 
+#' \code{$new()}. Individual methods are documented below.
+#' @section \code{$new()}:
+#' Initializes a new XGBoost deployment class using the 
+#' parameters saved in \code{p}, documented above. This method loads, cleans, and prepares data for
+#' generating predictions. \cr
+#' \emph{Usage:} \code{$new(p)}
+#' @section \code{$deploy()}:
+#' Generate new predictions and prepare the output dataframe. \cr
+#' \emph{Usage:} \code{$deploy()} 
+#' @section \code{$getPredictions()}:
+#' Return the grain and predictions for each class. \cr
+#' \emph{Usage:} \code{$getPredictions()} \cr
+#' @section \code{$getOutDf()}:
+#' Returns a dataframe containing the grain column, the top 3 probabilities for each row, 
```

Comment:
It's not, but you can get probabilities for all classes. Waiting on LIME for xgb explanations.